server/middleware/multer.js

```diff
@@ -0,0 +1,30 @@
+const multer = require("multer");
+const path = require("path");
+
+const storage = multer.diskStorage({});
+
+const upload = multer({
+  storage,
+  limit: { fileSize: 1000000 },
+  fileFilter: (req, file, cb) => {
+    checkFiletype(file, cb);
+  },
+});
+
+const checkFiletype = (file, cb) => {
+  const { originalname, mimetype } = file;
+  const validExtName = /jpeg|jpg|png|gif/;
+
+  const fileTypeCheck = validExtName.test(
+    path.extname(originalname).toLocaleLowerCase()
+  );
+  const mimeTypeCheck = validExtName.test(mimetype);
+
+  if (fileTypeCheck && mimeTypeCheck) {
+    return cb(null, true);
+  } else {
+    return cb(null, false);
+  }
+};
```

Comment:
Question: Any chance we can utilize `async/await` here? If not let me know!

Reply:
Yes,
 ```
fileTypeCheck = await validExtName.test(
    path.extname(originalname).toLocaleLowerCase()
  );
  const mimeTypeCheck = await validExtName.test(mimetype);
```
I missed it, well noted!